python-avd/pyavd/_eos_designs/shared_utils/utils.py

```diff
@@ -32,7 +33,17 @@ class UtilsMixin(Protocol):
     resolved_port_profiles_cache: dict[str, EosDesigns.PortProfilesItem] | None = None
     """Poor-mans cache to only resolve and deepmerge a port_profile once."""
 
-    def get_peer_facts(self: SharedUtilsProtocol, peer_name: str, required: bool = True) -> EosDesignsFacts | dict | None:
+    @cached_property
+    def switch_facts(self: SharedUtilsProtocol) -> EosDesignsFactsProtocol:
+        return self.get_peer_facts(self.hostname)
+
+    @overload
+    def get_peer_facts(self: SharedUtilsProtocol, peer_name: str, required: Literal[True] = True) -> EosDesignsFactsProtocol: ...
+
+    @overload
+    def get_peer_facts(self: SharedUtilsProtocol, peer_name: str, required: Literal[False]) -> EosDesignsFactsProtocol | None: ...
```

Comment:
do you really need this one? I would expect the function definition is enough

Reply:
Pyright complains without it.